rust/common-runtime/src/io.rs

```diff
@@ -1,12 +1,71 @@
-use std::collections::BTreeMap;
+use crate::{sync::ConditionalSync, CommonRuntimeError, Value, ValueKind};
+use common_ifc::Data;
+use common_macros::NewType;
+use common_protos::common;
+use std::collections::{BTreeMap, HashMap};
 
-use crate::{sync::ConditionalSync, Value, ValueKind};
+/// A wrapper type around the mapping of IO names
+/// for Common Modules.
+#[derive(NewType, Default, Clone, Debug)]
+pub struct DataIo(BTreeMap<String, Data<Value>>);
```

Comment:
It was confusing for me to read `DataIo` and `IoShape`. I think it might make more sense to call this `IoData`.